src/SimpleGraphs/generators/randgraphs.jl

```diff
@@ -967,6 +967,33 @@ function random_configuration_model(
     end
     return g
 end
+"""
+    uniform_tree(n)
+
+Generates a random labelled tree, drawn uniformly at random over the ``n^{n-2}`` such trees. A uniform word of length `n-2` over the alphabet `1:n` is generated (Prüfer sequence) then decoded. See also the `prufer_decode` function and [this page on Prüfer codes](https://en.wikipedia.org/wiki/Pr%C3%BCfer_sequence). 
+
+### Optional Arguments
+- `rng=nothing`: set the Random Number Generator.
+- `seed=nothing`: set the RNG seed.
+
+# Examples
+```jldoctest
+julia> uniform_tree(10)
+{10, 9} undirected simple Int64 graph
+```
+"""
+function uniform_tree(
+    n::Integer;
+    rng::Union{Nothing,AbstractRNG}=nothing,
+    seed::Union{Nothing,Integer}=nothing,
+)
+    n == 1 && return Graph(1, 0)
+    n == 2 && return path_graph(2)
+
+    rng = rng_from_rng_or_seed(rng, seed)
+    random_code = rand(rng, 1:n, n - 2)
```

Comment:
```suggestion
    n <= 1 && return Graph(n)
    n == 2 && return path_graph(n)

    rng = rng_from_rng_or_seed(rng, seed)
    random_code = rand(rng, Base.OneTo(n), n - 2)
```
This way, we can determine the eltype of the graph form the type of `n` - alternatively we could also add an optional argument that determines the eltype - but we rarely do this at least for now.

Reply:
Not sure, if we should also accept 0 vertex graphs here, or if we should rather fail in that case - probably the later, as `is_tree` would not be true in that case.

Reply:
about `1:n` the problem with that is, that
```julia
julia> typeof(1:UInt32(4))
UnitRange{Int64}

julia> typeof(1:UInt64(4))
UnitRange{UInt64}
```
so this is probably not what we want - so its better to either cast `1` to the correct type, or use `Base.OneTo`.

Reply:
Without the change proposed here, this function would also not be type stable if `n` is a `UInt64` for example. One can check this by calling using the macro `@code_warntype`:
```julia
@code_warntype uniform_tree(UInt64(2))
```
should some type instabilities.

For testing there is also the `@inferred` macro that helps for checking type stability: https://docs.julialang.org/en/v1/stdlib/Test/#Test.@inferred